widgets/src/image.rs

```diff
@@ -168,5 +173,14 @@ impl ImageRef {
             None
         }
     }
+
+    /// See [`Image::has_texture()`].
+    pub fn has_texture(&self) -> bool {
+        if let Some(inner) = self.borrow() {
+            inner.has_texture()
+        } else {
+            false
```

Comment:
When the internal ref cell is already mutably borrowed and this fails the options are:
- Return `false` (current behavior).
- Return Option or Result (exposing the fact that this is internally a ref cell).
- Simply blow (panic/unwrap/expect). This also exposes the fact that there is a ref cell involved but lets the user dev know about a borrowing issue (at runtime). So I would prefer this but it would not be consistent with the silent fails behavior in the rest of the widget.

None of the options are ideal, but they are the only options I can think of if there is a RefCell behind the scenes.